src/MOI/MOI_wrapper.jl

```diff
@@ -2801,21 +2818,34 @@ function MOI.get(model::Optimizer, attr::MOI.DualStatus)
     end
 end
 
+_update_cache(::Optimizer, data::Vector{Float64}, ::Any) = data
+
+function _update_cache(model::Optimizer, ::Nothing, f_p::F) where {F}
+    n = length(model.variable_info)
+    x = zeros(n)
+    ret = f_p(model.env, model.lp, x, 0, n-1)
+    _check_ret(model, ret)
+    return x
+end
+
+_get_cached_solution(model::Optimizer, data::Vector{Float64}, x) = data[column(model, x)]
```

Comment:
I don't think that adding this function improves the clarity of the code (given that it's only called once): rather, I would say that it mystifies it a bit when you call this function. I you want to keep it, I would remove the second argument (`data`), as it's always `model.variable_primal`.

Reply:
Agreed Just call `model.variable_primal[column(model, x)]` directly in `MOI.get` below.

Reply:
done